File: Examples_VI_CtCI_C/Example 14/Example14.c
#include <stdio.h>
// O (2^N) runtime.

int fib(int n) {
	if (n <= 0) return 0;
	else if(n == 1) return 1;
	return fib(n - 1) + fib (n - 2);
}

void allFib(int n) {
	int i;
	
	for (i = 0; i <= n; i++) {
		printf("%d: %d\n", i, fib(i));
	}	
}

int main() {
	int n;
	allFib(20);
}

File: Examples_VI_CtCI_Java/src/Example14.java

public class Example14 {
	// O (2^N) runtime.
	
	static void allFib(int n) {
		for (int i = 0; i <= n; i++) {
			System.out.println(i + ": " + fib(i));
		}
	}
	
	static int fib(int n) {
		if (n <= 0) return 0;
		else if (n == 1) return 1;
		return fib(n - 1) + fib(n - 2);
	}
	
	public static void main(String[] args) {
		allFib(20);
	}
}
